salt/renderers/pass.py

```diff
@@ -84,9 +94,13 @@ def _fetch_secret(pass_path):
     # The version of pass used during development sent output to
     # stdout instead of stderr even though its returncode was non zero.
     if proc.returncode or not pass_data:
-        log.warning("Could not fetch secret: %s %s", pass_data, pass_error)
+        if "pass_renderer_fail_hard" in __opts__ and __opts__["pass_renderer_fail_hard"]:
+            raise SaltRenderError("Unable to retrieve secret '{}' from the password store: {}".format(pass_path, stderr))
+        else:
+            log.warning("Could not fetch secret: %s %s", pass_data, pass_error)
         pass_data = pass_path
-    return pass_data.strip()
+    # only strip trailing whitespaces
+    return pass_data.rstrip('\r\n')
```

Comment:
According to the comment, you want to strip only trailing whitespaces, then shouldn't we use simply `rstrip()` here?

Passing `\r\n` won't strip trailing whitespaces in case our string is like `test   `.

Reply:
yeah we only want to strip newlines so the password can end in a ` `

Reply:
Ok :+1: let's fix the comment then to avoid confussion